api/openapi.yaml

```diff
@@ -353,6 +353,42 @@ paths:
           description: Endpoint not found
         '405':
           description: 'Invalid method, only GET,POST is allowed'
+
+  /bmc/creds:
+    parameters:
+      - in: query
+        name: targets
+        required: false
+        description: Comma separated list of XNames.  No query string results in fetching creds for all known BMCs.
+        schema:
+          $ref: '#/components/schemas/xname_list'
+      - in: query
+        name: type
+        required: false
+        description: Target component type.  A maximum of one type is allowed.  If no type is specified, all known BMC types are returned.
+        schema:
+          type: string
+          example: NodeBMC
+    get:
+      tags:
+        - creds
+      summary: Fetch controller login credentials for specified targets.
+      description: >-
+        Fetch controller login credentials for a specified targets.  Targets
+        are specified as a comma-separated list of xnames.
+      responses:
+        '200':
+          description: OK.  The data was succesfully set
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/creds_fetch_rsp'
+        '404':
```

Comment:
under what cases will a 404 be returned? if ANY xname isnt found in the list?

Reply:
IMO since there is the 'statuscode' field in the struct; it seems like you would always want to return the payload, but put a 4XX on some things as needed? thoughts?

Reply:
Good point.   My thinking was, let's filter out what doesn't make sense in the query params (AND of
xnames and types) and only return data for the ones that aren't filtered out, and the return code per
target reflects whether or not the Vault fetch worked.

At the very least I should probably document that better.    Otherwise there will need to be more code changes.

Reply:
I clarified this in the /bmc/creds GET description.